doc/sphinx/src/building.rst

```diff
@@ -1,23 +1,662 @@
-.. _building:
+Overview
+========
 
-Building Singularity-EOS
-=========================
+The ``singularity-eos`` build system is designed with two goals in mind
```

Comment:
somewhere in here we might want to say something about portability strategy through ports of call... even if we just link to ports of call documentation.

Reply:
Does the build system need to be able interpret `PORTABILITY_STRATEGY_KOKKOS` as implying `SINGULARITY_USE_KOKKOS`? (right now it's the reverse).